# fix arguments passing to main function

--- a/bin/pyflakes
+++ b/bin/pyflakes
@@ -1,4 +1,5 @@
 #!/usr/bin/python
 
+import sys
 from pyflakes.scripts.pyflakes import main
-main()
+main(sys.argv[1:])
--- a/pyflakes/scripts/pyflakes.py
+++ b/pyflakes/scripts/pyflakes.py
@@ -46,9 +46,8 @@
         print >> sys.stderr, '%s: no such file' % (filename,)
         return 1
 
-def main():
+def main(args):
     warnings = 0
-    args = sys.argv[1:]
     if args:
         for arg in args:
             if os.path.isdir(arg):
